src/Directory.Build.props

```diff
@@ -9,6 +9,14 @@
     <ErrorLog>$(MSBuildProjectFullPath).$([System.Guid]::NewGuid().ToString().Substring(0,8)).sarif</ErrorLog>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <PackageReadmeFile>README.md</PackageReadmeFile>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <None Include="README.md" Pack="true" PackagePath="\" />
+  </ItemGroup>
+
```

Comment:
```suggestion
  <PropertyGroup>
    <ShouldPackReadme Condition="'$(ShouldPackReadme)' == '' And '$(IsPackable)' == 'true' And Exists('$(MSBuildProjectDirectory)\README.md')">true</ShouldPackReadme>
    <PackageReadmeFile Condition="'$(ShouldPackReadme)' == 'true'>README.md</PackageReadmeFile>
  </PropertyGroup>

  <ItemGroup >
    <None Condition="'$(ShouldPackReadme)' == 'true' Include="README.md" Pack="true" PackagePath="\" />
  </ItemGroup>

```
NIT

Reply:
didn't test the above though so if you do take the suggestion it would be good to validate that it works 😄

Reply:
But we want it to fail if there isn't a readme file. Non-packable projects are already not affected.

Reply:
I see. If we want it to fail, usually it's best to have a target that runs post Pack that validates that a readme was added, and uses `<Error` task instead so that we can show a more custom message than just something around `some x file wasn't found`, but I don't want to block the PR on this so whatever you decide is fine.

Reply:
This seems clear enough?
```
D:\git\extensions\.dotnet\sdk\8.0.100-rc.2.23462.28\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.P
ack.targets(221,5): error NU5019: File not found: 'D:\git\extensions\src\Libraries\Microsoft.AspNetCore.AsyncState\READ
ME.md'. [D:\git\extensions\src\Libraries\Microsoft.AspNetCore.AsyncState\Microsoft.AspNetCore.AsyncState.csproj]
```

Reply:
Yeah, I think that's okay.